### PR TITLE
bpo-35224: Add What's new entry for evaluation order in dict comprehensions

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue35224

### CODE REVIEW

**Review by gvanrossum — APPROVED** [review]

File: Doc/whatsnew/3.8.rst [hunk 1]
```diff
@@ -349,6 +349,9 @@ Other Language Changes
   is ``-1``, and a suitable power of that inverse for other negative exponents.
   (Contributed by Mark Dickinson in :issue:`36027`.)
 
+* When dictionary comprehensions are evaluated, the key is now evaluated before
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.8.rst:352]: Please add also an entry for the bytecode changes. Authors of the bytecode proceeding libraries will read that section but may miss changes in this section.

Comment (pablogsal) [reply Doc/whatsnew/3.8.rst:352]: Good point, will update the PR soon.

**Review by serhiy-storchaka — APPROVED** [review]

File: Doc/whatsnew/3.8.rst [hunk 2]
```diff
@@ -1502,6 +1505,11 @@ CPython bytecode changes
   when awaiting a next item in an :keyword:`async for` loop.
   (Contributed by Serhiy Storchaka in :issue:`33041`.)
 
+* The :opcode:`MAP_ADD` now expects the value as the first element in the
+  stack and the key as the second element. This change was made so the key
+  is always evaluated before the value in dictionary comprehensions, as
+  porposed by :pep:`572`. (Contributed by Jörn Heissler in :issue:`35224`.)
```
Comment (gvanrossum) [inline Doc/whatsnew/3.8.rst:1511]: porposed -> proposed

Comment (pablogsal) [reply Doc/whatsnew/3.8.rst:1511]: Fixed in https://github.com/python/cpython/pull/14371

**Pull request merged**